src/Common/src/System/Security/AccessControl/PrivilegeNotHeldException.cs

```diff
@@ -10,7 +10,12 @@ namespace System.Security.AccessControl
 {
     [Serializable]
     [System.Runtime.CompilerServices.TypeForwardedFrom("mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")]
-    public sealed class PrivilegeNotHeldException : UnauthorizedAccessException, ISerializable
+#if ACCESSCONTROL_ASSEMBLY
+    public 
+#else
+    internal
+#endif
+        sealed class PrivilegeNotHeldException : UnauthorizedAccessException, ISerializable
```

Comment:
This is a little strange.  Do we allow this exception to propagate out of the assembly that this gets built into as internal?  If so, that'll lead to a developer writing code like:
```C#
catch (PrivilegeNotHeldException) { ... }
```
and seeing `PrivilegeNotHeldException` exceptions not being caught by it, because they're not unified and are actually two completely different exception types, albeit with the same name.

Reply:
I discussed this approach with @jkotas to reuse existent `Privilege` objects https://github.com/dotnet/corefx/issues/17164#issuecomment-439629537
Do you have better idea?

Reply:
Do we allow this exception to propagate out  into user code?

Reply:
Not in theory for this namespace. At the moment there is only one place where we use exception https://github.com/dotnet/corefx/pull/33645/files#diff-466c09e7d1aad9f0599e4a112618b7a4R248
Is your concern about future `PrivilegeNotHeldException` misuse?

Reply:
> Is your concern about future PrivilegeNotHeldException misuse?

The concern would be if we allowed an instance of this exception to escape this library and into calling code.  If two types are defined with the same namespace/name but in different assemblies, they're still fundamentally two completely different types.  So if we're talking about an exception, and an instance of one is thrown and the other is specified in a catch block, the instance will pass right through the catch block because, although it happens to be named the same thing, it's a completely different type.  So, if this code threw a PrivilegeNotHeldException (this library's internal exception type of that name), and calling code had a catch block for PrivilegeNotHeldException (the public type exposed from another library), that catch block would not catch this exception, and in fact other than `catch (Exception)` to broadly catch all exception types, there'd be no specific way for the calling code to catch just this exception type, because it's internal.

That said, that concern isn't an issue if the exception never escapes this library, and that appears to be the case, so my concern doesn't apply.

Reply:
> That said, that concern isn't an issue if the exception never escapes this library, and that appears to be the case, so my concern doesn't apply.

Confirm "today", I'm aware that if someone in future "throw" this exception inside this project will be a problem.